crates/bin/node/src/grpc_service.rs

```diff
@@ -491,4 +563,24 @@ impl Node for GrpcState {
             info: node_info_str,
         }))
     }
+
+    /// acknowledge is for the client to say he successfully stored the quote_id
+
+    async fn acknowledge(
+        &self,
+        ack_request: Request<AcknowledgeRequest>,
+    ) -> Result<Response<AcknowledgeResponse>, Status> {
+        let method = ack_request.get_ref().method.clone();
+        let request_hash = ack_request.get_ref().request_hash.clone();
+
+        let cache_key = format!("{:?}{}", method, request_hash);
+
+        // check if the request is already in the cache, if so, remove it
+        let exist = self.response_cache.get(&cache_key.clone());
+        if exist.is_some() {
+            self.response_cache.remove(&cache_key);
+            return Ok(Response::new(AcknowledgeResponse {}));
+        }
+        Err(Status::not_found("Request is not found in the cache"))
```

Comment:
You can return success anyway rather than 404.
The goal of removing the element has been achieved